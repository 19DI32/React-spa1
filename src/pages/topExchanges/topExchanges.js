import React from 'react';
import './topExchanges.css';

export const TopExchanges = (props)=>{
    return(
        <div>
            <h2>TopExchanges</h2>
            <TopExchangesData/>
        </div>
    )
}

export class TopExchangesData extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isLoading:true,
            data:[]
        };
    }
    componentDidMount() {
        fetch('https://min-api.cryptocompare.com/data/top/exchanges?fsym=BTC&tsym=USD').then(data=>data.json()).then((data)=> {
            console.log(data.Data);
            this.setState({
                isLoading:false,
                data:data.Data
            })
        })
    }
    render () {
        if(this.state.isLoading) {
        return (
            <div>
                <h3>Loading...</h3>

            </div>
        )
    }
        else{
            return (
            <div>
               {this.state.data.map(elem =><ExchangesCard exchange = {elem.exchange} fromSymbol = {elem.fromSymbol} toSymbol = {elem.toSymbol} volume = {elem.volume24hTo}/>)}    
            </div>)
    }

}
}

function ExchangesCard(props) {
    return (
        <div className = "ExchangesCard">
            <h3>{"Exchange: "+ props.exchange}</h3>
            <h4>{"From Symbol: "+props.fromSymbol}</h4>
            <h4>{"To Symbol: "+props.toSymbol}</h4>
            <h4>{"volume: "+ props.volume}</h4>
        </div>
    )
}
//export default TopExchanges;