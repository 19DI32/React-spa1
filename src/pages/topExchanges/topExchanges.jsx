import React from 'react';
import { shallow } from 'enzyme';
import TopExcganges from './topExchanges';

test('Coins render correctly', () => {
    const component = shallow(<TopExcganges />);
    expect(component).toMatchSnapshot();
  });