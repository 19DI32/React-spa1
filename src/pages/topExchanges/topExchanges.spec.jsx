import React from 'react';
import { shallow } from 'enzyme';
import {TopExchanges, TopExchangesData} from './topExchanges';

test('Tops render correctly', () => {
    const component = shallow(<TopExchanges />);
    expect(component).toMatchSnapshot();
  });

test("Check loading", ()=> {
  const component = shallow(<TopExchangesData/>);
  expect(component.state(['isLoading'])).toEqual(true);
})