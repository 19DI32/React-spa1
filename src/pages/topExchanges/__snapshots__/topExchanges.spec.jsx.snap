// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tops render correctly 1`] = `ShallowWrapper {}`;
